src/searchengine/nova3/helpers.py

```diff
@@ -87,9 +87,9 @@ def entity2char(m):
     return re.sub(r'&#x(\w+);', lambda x: chr(int(x.group(1), 16)), t)
 
 
-def retrieve_url(url):
+def retrieve_url(url, custom_headers={}):
     """ Return the content of the url page as a string """
-    req = urllib.request.Request(url, headers=headers)
+    req = urllib.request.Request(url, headers={**headers, **custom_headers})
```

Comment:
How is it supposed to work in case when same keys are in `headers` and `custom_headers`?

Reply:
As @Chocobo1 says, the `custom_headers` value overwrites the `headers` value, I totally agree it's the plugin responsibility if the overwrite any  OOT header.

 I don't think there is any use case where a plugin developer should overwrite the user agent, but if they want to do it, they should be able to do so